### PR TITLE
PAN-OS EDL enhancements

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19753

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [ ] Documentation - soon
- [x] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/blob/ca9200efb9b6bf8e05fdd108ada5ced81922bd1b/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.yml)
- [ ] [CHANGELOG](https://github.com/demisto/content/blob/ca9200efb9b6bf8e05fdd108ada5ced81922bd1b/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/CHANGELOG.md)

### CODE REVIEW

**Comment by bakatzir** [comment]
@kirbles19 can you also review the warning/error messages I give the users?
https://github.com/demisto/content/blob/6d4f869d70dd996ad6b52f2156d4b23011f4bbe3/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py
lines:
510-511
515-516

File: Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py [hunk 1]
```diff
@@ -232,28 +246,47 @@ def edl_update():
...
-            'Contents': md,
-            'ContentsFormat': formats['markdown']
-        })
+def edl_update_internal_list_command():
+    """
+        Updates the instance context with the list name and items given
+    """
+    list_name = demisto.args().get('list_name')
+    list_items = argToList(demisto.args().get('list_items'))
+    add = demisto.args().get('add_or_remove') == 'add'
```
Comment (yuvalbenshalom) [inline Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py]: please check that you don't get anything else than "add" or "remove"
I you don't want to remove if you got a bad value.

File: Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py [hunk 2]
```diff
@@ -232,28 +246,47 @@ def edl_update():
...
+    """
+    Updates the instance context with the list name and items given
+    Overrides external file path with internal list
+    """
+    file_path = demisto.args().get('file_path')
+    if DOCUMENT_ROOT:
+        file_path = os.path.join(DOCUMENT_ROOT, file_path)
+    list_name = demisto.args().get('list_name')
+    list_items = argToList(demisto.args().get('list_items'))
+    add = demisto.args().get('add_or_remove') == 'add'
```
Comment (yuvalbenshalom) [inline Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py]: please check that you don't get anything else than "add" or "remove"
I you don't want to remove if you got a bad value.

**Comment by bakatzir** [comment]
@yuvalbenshalom 
asked alex to review the error string as well.
Screen Shot 2019-10-29 at 15 56 55

**Comment by kirbles19** [comment]
@bakatzir Done.

File: Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py [hunk 3]
```diff
@@ -474,16 +511,16 @@ def edl_compare_command():
...
                               headers=[file_path.rsplit('/')[-1]])
     if unique_internal:
-        md += '### Warning: Internal list has values which are not in the external file.\n'
-        md += '#### Please check who has writing permissions to the external file.\n'
+        md += '### Warning: Internal list contains values that are not in the external file.\n'
+        md += '#### If these changes are unexpected, check who has permission to write to the external file.\n'
         md += tableToMarkdown('', list(unique_internal), headers=[list_name])
     if len(md) == 0:
-        md = 'Internal list and External file have the same values'
+        md = 'Internal list and External file have the same values.'
```
Comment (kirbles19) [inline Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py]: external

**Pull request merged**